src/Dapr.Jobs/Extensions/DaprJobsServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,103 @@
+﻿// ------------------------------------------------------------------------
+// Copyright 2024 The Dapr Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//     http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ------------------------------------------------------------------------
+
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.DependencyInjection.Extensions;
+
+namespace Dapr.Jobs.Extensions;
+
+/// <summary>
+/// Contains extension methods for using Dapr Jobs with dependency injection.
+/// </summary>
+public static class DaprJobsServiceCollectionExtensions
+{
+    /// <summary>
+    /// Adds Dapr Jobs client support to the service collection.
+    /// </summary>
+    /// <param name="serviceCollection">The <see cref="IServiceCollection"/>.</param>
+    /// <returns></returns>
+    public static IServiceCollection AddDaprJobsClient(this IServiceCollection serviceCollection)
+    {
+        ArgumentNullException.ThrowIfNull(serviceCollection, nameof(serviceCollection));
+
+        //Register the IHttpClientFactory implementation
+        serviceCollection.AddHttpClient();
+
+        serviceCollection.TryAddSingleton(serviceProvider =>
+        {
+            var httpClientFactory = serviceProvider.GetRequiredService<IHttpClientFactory>();
+
+            var builder = new DaprJobsClientBuilder();
+            builder.UseHttpClientFactory(httpClientFactory);
+
+            return builder.Build();
+        });
+
+        return serviceCollection;
+    }
+
+    /// <summary>
+    /// Adds Dapr Jobs client support to the service collection.
+    /// </summary>
+    /// <param name="serviceCollection">The <see cref="IServiceCollection"/>.</param>
+    /// <param name="configure">Optionally allows greater configuration of the <see cref="DaprJobsClient"/>.</param>
+    public static IServiceCollection AddDaprJobsClient(this IServiceCollection serviceCollection, Action<DaprJobsClientBuilder>? configure)
+    {
+        ArgumentNullException.ThrowIfNull(serviceCollection, nameof(serviceCollection));
```

Comment:
Similar to above, this is just a specialization of the overload below?

Reply:
This is more keeping with the style of the how the `DaprClient` is currently built via the ASP.NET Core package in that the configuration can reflect only the `DaprJobsClientBuilder` or also include the `IServiceProvider` introduced in #1289 in case anything else needs to be injected as part of the registration (e.g. configuration or external secrets).